hack/update-codegen.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+ROOT_DIR=$(realpath $(dirname "${BASH_SOURCE[0]}")/..)
+
+TOOLS_DIR="${ROOT_DIR}"/hack/tools
+TOOLS_BIN_DIR="${TOOLS_DIR}/bin"
+
+CONTROLLER_TOOLS_VERSION="v0.11.1"
+
+CODEGEN_PKG=${CODEGEN_PKG:-$(cd "${ROOT_DIR}"; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../../k8s.io/code-generator)}
+
+KWOK_PROJECT="sigs.k8s.io/kwok"
+KWOK_API_PACKAGE="${KWOK_PROJECT}/pkg/apis"
+
+function codegen() {
+    echo "Update codegen"
+
+    bash "${CODEGEN_PKG}"/generate-internal-groups.sh \
+        "deepcopy,defaulter,conversion" \
+        "${KWOK_PROJECT}"/pkg/apis \
+        "${KWOK_PROJECT}"/pkg/apis \
+        "${KWOK_PROJECT}"/pkg/apis \
```

Comment:
Why is it repeated 3 times here?

Reply:
it is the args of `generate-internal-groups.sh`, means:
1. output pkg
2. internal API pkg
3. external API pkg

as kwok only have one API pkg, and will put the generated files under the same dir,
the 3 args is using the same one